### PR TITLE
Full source bootstrap for rust 1.29 🎉 

### PR DESCRIPTION
**Pull request by Elizafox**
This is the first step in a full bootstrap chain for Rust 1.29 through 1.74. This uses mrustc to do the initial bootstrap from pure source. Additional bootstrapping work will be followed up in future PR's.

#### For new package PRs only
- [ ] This PR is marked as fixing a pre-existing package request bug (part of chainguard-dev/internal#3355)
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ ] REQUIRED - The version of the package is still receiving security updates (**EXCEPTION**: bootstrapping)
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

### CODE REVIEW

**Comment by Elizafox** [comment]
Here is a screenshot of it working:

Screenshot 2023-12-06 at 4 09 09 PM

**Comment by kaniini** [comment]
It seems that the job was reaped on both hosts, which indicates that perhaps the runner's resources exceeded what they were allowed.  Maybe we can use a smaller parallelism than nproc?

File: rust-1.29.yaml [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+      export CPPFLAGS="${{vars.CPPFLAGS}}"
+      export CXXFLAGS="${{vars.CXXFLAGS}}"
+
+      export OPENSSL_NO_VENDOR=1
+      export RUST_BACKTRACE=1
+      DESTDIR=${{targets.destdir}} python3 ./x.py install --jobs $(nproc)
+
+  - uses: strip
+
+  # delete uneeded files eg uninstalltion
```
Comment (kaniini) [inline rust-1.29.yaml:226]:
```suggestion
  # delete uneeded files such as uninstallation scripts
```

File: rust-1.29.yaml [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+      # Remove vendored LLVM stored in /home/build, not needed as rustc is statically linked
+      rm -r ${{targets.destdir}}/home
+
+      rm ${{targets.destdir}}/usr/lib/rustlib/components
+      rm ${{targets.destdir}}/usr/lib/rustlib/install.log
+      rm ${{targets.destdir}}/usr/lib/rustlib/rust-installer-version
+      rm ${{targets.destdir}}/usr/lib/rustlib/uninstall.sh
+      rm ${{targets.destdir}}/usr/lib/rustlib/manifest-*
+
+  # rustbuild always installs copies of the shared libraries to /usr/lib,
```
[the remark this answers is not in the document]
Comment (Elizafox) [reply rust-1.29.yaml:237]: These files are actually generated by Rust itself and this rm does not fail on ARM either. These should probably be stripped out. I took this from the existing Rust build file, fwiw.

File: rust-1.29.yaml (hunk 1 above)
Comment (Elizafox) [reply rust-1.29.yaml:226]: This is also from the original Rust build. Looks good to me.

**Comment by Elizafox** [comment]
Okay I did an update and then it failed. Maybe it's when the runners are busy?

**Comment by Elizafox** [comment]
The CI passes now! This can be merged now. /CC @kaniini

**Review by kaniini — DISMISSED** [review]

**Comment by Elizafox** [comment]
This will be superseded.

**Pull request closed without merging**